packages/env/lib/build-docker-compose-config.js

```diff
@@ -262,6 +273,15 @@ module.exports = function buildDockerComposeConfig( config ) {
 				volumes: testsMounts,
 			},
 			cli: {
+				build: {
+					context: '.',
+					dockerfile: 'CLI.Dockerfile',
```

Comment:
I'm noticing we aren't using the Dockerfile for the tests instance (both before for WordPress and now for CLI). Should we add the dockerfile to those too?

Reply:
Interestingly enough, the tests instance _actually still work anyway_! I wanted to understand why that is the case, so, I've spent some time looking into it. As far as I can tell though, it should only work in the case where both environments have the same `phpVersion`.

- When you use `image` and `build` together in `docker-compose.yml`, Docker publishes the image that it builds in your local registry. When the `tests-wordpress` image uses the `wordpress` image, it's using the one that the `wordpress` service just created!
- Since we use `dockerCompose.pullAll()`, we're replacing the images and so this is fine!

This isn't _that_ strange of a workflow, but, normally you would use your own `image:` property instead of replacing the one you're trying to pull down in the first place. This is only a problem if you try to `pull` images since the ones you're referencing probably don't really exist on the remote.

---

To our purpose, though, yes, you're right. It doesn't really work! If the PHP versions are different then the test instances will pull a fresh image instead of the one that was built by the development services. With that in mind, we're going to need four different Dockerfiles to generate a unique image for each of them. As a nice bonus, by removing the `image:` properties, we can avoid this weird contention that may even break other Docker Compose setups using the same WordPress images on the user's computer 💀 

Great catch!